File: benchmark/c/cpw/add_two_numbers_without_using_arithmetic_operators/add_two_numbers_without_using_arithmetic_operators.c


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>



int add_two_numbers_without_using_arithmetic_operators ( int x, int y ) ;
int add_two_numbers_without_using_arithmetic_operators ( int x, int y ) {
  while ( y != 0 ) {
    int carry = x & y;
    x = x ^ y;
    y = carry << 1;
  }
  return x;
}

File: translation_result/else_bms/success/c/cpw/add_two_numbers_without_using_arithmetic_operators/gemini/CAPR/add_two_numbers_without_using_arithmetic_operators.rs

fn add_two_numbers_without_using_arithmetic_operators(mut x: i32, mut y: i32) -> i32 {
    while y != 0 {
        let carry = (x & y) << 1;
        x = x ^ y;
        y = carry;
    }
    x
}
